websites/N/nadeshiko/presence.ts

```diff
@@ -0,0 +1,246 @@
+const presence = new Presence({
+	clientId: "1332497160406695947",
+}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/tDAXNRP.png",
+}
+
+async function getStrings() {
+	return presence.getStrings({
+		browsing: "general.browsing",
+		search: "general.search",
+		viewHome: "general.viewHome",
+		viewPage: "general.viewPage",
+		viewing: "general.viewing",
+		viewAPage: "general.viewAPage",
+		viewProfile: "general.viewProfile",
+		viewAProfile: "general.viewAProfile",
+		buttonViewPage: "general.buttonViewPage",
+		buttonViewProfile: "general.buttonViewProfile",
+	});
+}
+
+let strings: Awaited<ReturnType<typeof getStrings>>,
+	oldLang: string = null;
+
+const gameNames: Record<string, string> = {
+	network: "Network",
+	bedwars: "Bed Wars",
+	duels: "Duels",
+	skywars: "SkyWars",
+	arcade: "Arcade",
+	blitz: "Blitz",
+	buildbattle: "Build Battle",
+	classic: "Classic",
+	copsandcrims: "Cops and Crims",
+	megawalls: "Mega Walls",
+	murdermystery: "Murder Mystery",
+	pit: "Pit",
+	smashheroes: "Smash Heroes",
+	tntgames: "TNT Games",
+	uhc: "UHC",
+	warlords: "Warlords",
+	woolgames: "Wool Games",
+	fishing: "Fishing",
+},
+	achievementsGameNames: Record<string, string> = {
+		overall: "Overall",
+		legacy: "Legacy",
+		general: "General",
+		arcade: "Arcade",
+		arena: "Arena Brawl",
+		bedwars: "Bed Wars",
+		blitz: "Blitz",
+		buildbattle: "Build Battle",
+		copsandcrims: "Cops and Crims",
+		duels: "Duels",
+		housing: "Housing",
+		megawalls: "Mega Walls",
+		murdermystery: "Murder Mystery",
+		paintball: "Paintball",
+		pit: "Pit",
+		quakecraft: "Quakecraft",
+		skyblock: "SkyBlock",
+		skywars: "SkyWars",
+		smashheroes: "Smash Heroes",
+		speeduhc: "Speed UHC",
+		tkr: "Turbo Kat Racers",
+		tntgames: "TNT Games",
+		uhc: "UHC",
+		vampirez: "VampireZ",
+		walls: "Walls",
+		warlords: "Warlords",
+		woolgames: "Wool Games",
+		easter: "Easter",
+		halloween: "Halloween",
+		holiday: "Holiday",
+		summer: "Summer",
+	};
+
+presence.on("UpdateData", async () => {
+	const [
+		languageSetting,
+		privacyModeSetting,
+		showSmallImagesSetting,
+		showButtonsSetting,
+	] = await Promise.all([
+		presence.getSetting<string>("language").catch(() => "en"),
+		presence.getSetting<boolean>("privacyMode"),
+		presence.getSetting<boolean>("showSmallImages"),
+		presence.getSetting<boolean>("showButtons"),
+	]);
+
+	if (oldLang !== languageSetting) {
+		oldLang = languageSetting;
+
+		strings = await getStrings();
+	}
```

Comment:
The current implementation of `getStrings` no longer needs the `oldLang` hack to work efficiently. Instead, just embed the `presence.getStrings` directly here
```ts
const strings = await presence.getStrings(...)
```